File: server/makc2022--nodejs--nestjs/src/layer3/nosql-mongo/sample/entities/dummy-many-to-many/dummy-many-to-many-entity.loader.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

import { nameof } from 'src/layer1/common/common-functions';
import { EntityLoader } from 'src/layer1/entity/entity-loader';
import { DummyManyToManyEntityObject } from './dummy-many-to-many-entity.object';

export class DummyManyToManyEntityLoader extends EntityLoader<DummyManyToManyEntityObject> {
  /** @inheritdoc */
  constructor(entityObject?: DummyManyToManyEntityObject) {
    super(entityObject ?? ({} as DummyManyToManyEntityObject));
  }

  /** @inheritdoc */
  override load(
    entityObject: DummyManyToManyEntityObject,
    loadableProperties?: Set<string>
  ): Set<string> {
    const result = super.load(entityObject, loadableProperties);

    if (result.has(nameof<DummyManyToManyEntityObject>('id'))) {
      this.entityObject.id = entityObject.id;
    }

    if (result.has(nameof<DummyManyToManyEntityObject>('name'))) {
      this.entityObject.name = entityObject.name;
    }

    return result;
  }

  /** @inheritdoc */
  protected override createAllPropertiesToLoad(): Set<string> {
    return new Set<string>([
      nameof<DummyManyToManyEntityObject>('id'),
      nameof<DummyManyToManyEntityObject>('name'),
    ]);
  }
}

File: server/makc2022--dotnet/Makc2022.Layer3.Sql.Sample/Entities/DummyManyToMany/DummyManyToManyEntityLoader.cs
// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2022.Layer3.Sql.Sample.Entity;

namespace Makc2022.Layer3.Sql.Sample.Entities.DummyManyToMany
{
    /// <summary>
    /// Загрузчик сущности "DummyManyToMany".
    /// </summary>
    public class DummyManyToManyEntityLoader : EntityLoader<DummyManyToManyEntityObject>
    {
        #region Constructors

        /// <inheritdoc/>
        public DummyManyToManyEntityLoader(DummyManyToManyEntityObject? entityObject = null)
            : base(entityObject ?? new DummyManyToManyEntityObject())
        {
        }

        #endregion Constructors

        #region Public methods

        /// <inheritdoc/>
        public sealed override HashSet<string> Load(
            DummyManyToManyEntityObject entityObject,
            HashSet<string>? loadableProperties = null)
        {
            var result = base.Load(entityObject, loadableProperties);

            if (result.Contains(nameof(EntityObject.Id)))
            {
                EntityObject.Id = entityObject.Id;
            }

            if (result.Contains(nameof(EntityObject.Name)))
            {
                EntityObject.Name = entityObject.Name;
            }

            return result;
        }

        #endregion Public methods

        #region Protected methods

        /// <inheritdoc/>
        protected override HashSet<string> CreateAllPropertiesToLoad()
        {
            return new HashSet<string>
            {
                nameof(EntityObject.Id),
                nameof(EntityObject.Name)
            };
        }

        #endregion Protected methods
    }
}
